src/screen.rs

```diff
@@ -60,23 +76,24 @@ impl Interfaceable for PrettyPrompt {
         println!("{}", string);
     }
 
-    fn prompt(&mut self) -> String {
+    fn prompt(&mut self) -> Result<String, MaeveError> {
         let readline = match self.editor.readline(">> ") {
             Ok(line) => {
                 if self.history {
                     self.editor.add_history_entry(&line);
                 }
-                return line;
+                line
             }
             Err(_) => String::from("quit"),
         };
-        if confirm_history() {
-            match self.editor.save_history(".history.txt") {
-                Ok(_) => (),
-                Err(_) => return String::from("Error writing .history.txt."),
-            };
+        match (
```

Comment:
Should only run if history is true

Reply:
This shouldn't be in this branch...

Reply:
The OpenOptions method actually checks for this inside of the confirm_history() function. It will only create a new file if one doesn't already exist by default, as opposed to what we were doing before which I think was the same thing but manually.